File: 2015/ruby/day4/part1.rb
#!/usr/bin/ruby
require 'digest'

input = 'ckczppom'
number = 1
hash = Digest::MD5.hexdigest(input + number.to_s)

until hash.start_with?('00000')
    number += 1
    hash = Digest::MD5.hexdigest(input + number.to_s)
end

puts number

File: 2016/python/day5/part1.py
#!/usr/bin/python
import hashlib

door = 'ffykfhsq'
index = 0
password = ''

while len(password) < 8:
    digest = hashlib.md5(door + str(index)).hexdigest()
    index += 1

    if digest.startswith('00000'):
        password += digest[5]

print password
